src/modules/ekf2/EKF2.cpp

```diff
@@ -225,20 +225,23 @@ int EKF2::print_status()
 	return 0;
 }
 
-template<typename Param>
-void EKF2::update_mag_bias(Param &mag_bias_param, int axis_index)
+template<typename Param1, typename Param2>
+void EKF2::update_mag_bias(Param1 &mag_bias_param, Param2 &mag_bias_var_param, int axis_index)
 {
 	if (_valid_cal_available[axis_index]) {
 
 		// calculate weighting using ratio of variances and update stored bias values
-		const float weighting = constrain(_param_ekf2_magb_vref.get() / (_param_ekf2_magb_vref.get() +
+		const float weighting = constrain(mag_bias_var_param.get() / (mag_bias_var_param.get() +
 						  _last_valid_variance[axis_index]), 0.0f, _param_ekf2_magb_k.get());
 		const float mag_bias_saved = mag_bias_param.get();
 
 		_last_valid_mag_cal[axis_index] = weighting * _last_valid_mag_cal[axis_index] + mag_bias_saved;
+		const float new_ref_var = weighting * _last_valid_variance[axis_index] + (1.f - weighting) * mag_bias_var_param.get();
```

Comment:
Reuse `AlphaFilter<float>[]` for `_last_valid_variance`  https://github.com/PX4/ecl/blob/99aafbaa18986342baf2f69c4fc507a76a8d76b2/AlphaFilter/AlphaFilter.hpp#L93 ?

Reply:
Well, it isn't worth instantiating a filter just for a single update per flight

Reply:
Ah I see 👍